src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/ExpressionLambdaRewriter.cs

```diff
@@ -202,7 +202,12 @@ private BoundExpression VisitExpressionWithoutStackGuard(BoundExpression node)
                 case BoundKind.DelegateCreationExpression:
                     return VisitDelegateCreationExpression((BoundDelegateCreationExpression)node);
                 case BoundKind.FieldAccess:
-                    return VisitFieldAccess((BoundFieldAccess)node);
+                    var fieldAccess = (BoundFieldAccess)node;
+                    if (fieldAccess.FieldSymbol.IsCapturedFrame)
```

Comment:
It isn't clear that this is correct in the case of nested lambdas, as the inner frame may have different values on each invocation of the outer lambda. But perhaps this is OK because it is a constant each time the *inner* lambda is evaluated. It would be helpful to ensure we have coverage for this interaction.